catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.java

```diff
@@ -51,9 +55,10 @@ protected void onCreate(Bundle savedInstanceState) {
 		titleActionBar = null;
 		returnToProjectsList = false;
 		returnByPressingBackButton = false;
-		Thread.setDefaultUncaughtExceptionHandler(new BaseExceptionHandler());
-		Utils.checkIfCrashRecoveryAndFinishActivity(this);
+		Thread.setDefaultUncaughtExceptionHandler(new BaseExceptionHandler(this));
+		checkIfCrashRecoveryAndFinishActivity(this);
 		getActionBar().setDisplayHomeAsUpEnabled(true);
+		CrashReporter.sendUnhandledCaughtException();
```

Comment:
this is really obvuscating whats going on here, it says sendUnhandledCaughtException and is called EVERY time on create of the activity.
why not move the whole thing to checkIfCrashRecovery or into exceptionHandler or somewhere else...